src/main/java/org/folio/event/handler/HoldingCreateAsyncRecordHandler.java

```diff
@@ -58,7 +58,7 @@ protected Future<Void> processInventoryCreationEvent(ResourceEvent resourceEvent
     var tenantIdFromEvent = resourceEvent.getTenant();
     return dbClient.getPgClient()
       .withTrans(conn -> {
-        var updatedHeaders = HeaderUtils.copyHeadersAndUpdatedTenant(centralTenantId, headers);
+        var updatedHeaders = HeaderUtils.prepareHeaderForTenant(centralTenantId, headers);
```

Comment:
Note to future me:

- Kafka headers are not case-insensitive: 
![image](https://github.com/user-attachments/assets/9612dc78-4bb5-4da5-88e9-c1d65a07c027)

Thus any implementation or manipulation of Kafka headers must be performed using a case-insensitive Map implementation, such as `org.apache.commons.collections4.map.CaseInsensitiveMap<K, V>`